File: python/2015/hello-my-name-is-doris-add.py

from movies import MovieRepository

# Create a new movie repository using the default configuration
movies = MovieRepository()

# Add Hello, My Name Is Doris to the database
movies.insert(
    title="Hello, My Name Is Doris", 
    year=2015, 
    plot="A self-help seminar inspires a sixty-something woman to romantically pursue her younger co-worker.", 
    rating=6.7
)

# Confirm that the movie was added
movie = movies.select(
    title="Hello, My Name Is Doris", 
    year=2015
)
if movie:
    # The movie was found
    print(f"Movie found: {movie}")
else:
    # The movie was not found
    print("Movie not found")
    

File: csharp/2015/hello-my-name-is-doris-add.cs
using System;
using System.Threading.Tasks;
using Amazon.DynamoDBv2;
using Amazon.DynamoDBv2.Model;
using System.Collections.Generic;

namespace MovieDatabase
{
    class Program
    {
        static async Task Main(string[] args)
        {
            // Create a MovieRepository instance
            var movies = new MovieRepository();

            // Add "Hello, My Name Is Doris" to the database
            // This demonstrates how to insert a new item into DynamoDB
            await movies.InsertAsync(
                title: "Hello, My Name Is Doris",
                year: 2015,
                plot: "A self-help seminar inspires a sixty-something woman to romantically pursue her younger co-worker.",
                rating: 6.7
            );

            // Confirm that the movie was added by retrieving it
            var movie = await movies.SelectAsync(
                title: "Hello, My Name Is Doris",
                year: 2015
            );

            if (movie != null)
            {
                // The movie was found
                Console.WriteLine($"Movie found: {movie}");
            }
            else
            {
                // The movie was not found
                Console.WriteLine("Movie not found");
            }
        }
    }
}
